src/daft-catalog/src/impls/memory.rs

```diff
@@ -0,0 +1,369 @@
+use std::sync::{Arc, RwLock};
+
+use common_error::DaftResult;
+use daft_context::get_context;
+use daft_core::prelude::SchemaRef;
+use daft_dsl::LiteralValue;
+use daft_logical_plan::{ops::Source, InMemoryInfo, LogicalPlan, LogicalPlanBuilder, SourceInfo};
+use daft_micropartition::partitioning::{MicroPartitionSet, PartitionSet};
+use indexmap::IndexMap;
+
+use crate::{
+    error::{CatalogError, CatalogResult},
+    Catalog, Identifier, Table, TableRef,
+};
+
+type NamespaceTableMap = IndexMap<Option<String>, IndexMap<String, Arc<MemoryTable>>>;
+
+#[derive(Clone, Debug)]
+/// A catalog entirely stored in-memory.
+///
+/// Supports tables without namespaces or with a single level namespace.
```

Comment:
interesting that rustfmt didn't bark

Reply:
Oh are there better formatting options for this

Reply:
🤷  maybe it's just my editor that lints this, https://github.com/rust-lang/style-team/issues/72

Reply:
Ah yeah I'll move it.